### DISCUSSION TITLE
Anyway to test if the HAN-port is activated

### DISCUSSION THREAD

**Discussion by ancan** [UtilitechAS/amsreader-firmware#385] [Q&A] [answered]
I slapped together my own hardware with a Wemos D1 mini and a M-bus adapter, but can't get any readings and HAN-status is red. I understand the M-bus uses 24V on it's bus, but measuring I get 0V. Does this mean the HAN-port is still not activated? Activation was requested days ago.

**Comment by ArnieO** [comment, marked as answer]
It sounds like a non-activated HAN port.
Which country are you in? 
This might be of interest: https://amsleser.no/blog/post/20-vattenfall-opening-the-han-port
But to answer your question: You will be able to see with a multimeter that the voltage is 24V most of the time, then fluctuates for a duration of around 1 second each time data arrives from the meter..

**Reply by ancan** [reply]
It turns out there is voltage on the port, but the moment I connect it to the M-bus adapter it drops to zero, so I guess there's something wrong in my setup. Sockets I connect to the port are marked "MB A <<->>", and TXD/GND/VIN connects to the Wemos.
(I'm in Norway)

**Reply by ArnieO** [reply]
The experience with those adapters is "varying", please see if you can find useful information in this thread: 
https://www.hjemmeautomasjon.no/forums/topic/2873-lokal-lesing-av-han-the-easy-way-tm/

**Reply by ancan** [reply]
Thanks! I did find it.

"TSS721 har fire tilkoblinger merket MB_A, MB_A, MB_B og MB_B (!). Det viser seg at det egentlig kun er to tilkoblinger, men at det sansynligvis er lagt opp til daisy-chaining siden MB_A er koblet sammen og MB_B er koblet sammen. **HAN skal kobles til med en ledning mot MB_A (samme hvilken) og en ledning mot MB_B** (samme hvilken)."

So moving one cable to "B", and it started working.

Edit: For non-norwegian speakers looking for the same answer, even though the TSS721 has four ports marked MB_A, MB_A, MB_B and MB_B, it's not that it has two channels "A" and "B". Both A are electrically the same, and both B are electrically the same, and the HAN-port should be connected with one wire on A and one wire on B. Which goes where is irellevant.
